x-pack/plugins/observability_ai_assistant/public/utils/get_timeline_items_from_conversation.tsx

```diff
@@ -6,18 +6,40 @@
  */
 import { i18n } from '@kbn/i18n';
 import type { AuthenticatedUser } from '@kbn/security-plugin/common';
+import { isEmpty, omitBy } from 'lodash';
 import React from 'react';
 import { v4 } from 'uuid';
 import { Message, MessageRole } from '../../common';
 import type { ChatTimelineItem } from '../components/chat/chat_timeline';
 import { RenderFunction } from '../components/render_function';
 import type { ObservabilityAIAssistantChatService } from '../types';
 
-function convertFunctionParamsToMarkdownCodeBlock(object: Record<string, string | number>) {
-  return `
-\`\`\`
-${JSON.stringify(object, null, 4)}
-\`\`\``;
+function convertMessageToMarkdownCodeBlock(message: Message['message']) {
+  let value: object;
+
+  if (!message.name) {
+    const name = message.function_call?.name;
+    const args = message.function_call?.arguments
+      ? JSON.parse(message.function_call.arguments)
+      : undefined;
+
+    value = {
+      name,
+      args,
+    };
+  } else {
+    const content = message.content ? JSON.parse(message.content) : undefined;
+    const data = message.data ? JSON.parse(message.data) : undefined;
+    value = omitBy(
+      {
+        content,
+        data,
+      },
+      isEmpty
+    );
+  }
+
+  return `\`\`\`\n${JSON.stringify(value, null, 2)}\n\`\`\``;
 }
```

Comment:
this serializes both function suggestions (`{ function_call: { ... } } `) and function responses (`{ name: ..., content: ...}`)